### PR TITLE
Fold api.Point into api.DOMPoint

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR folds the `Point` API into `DOMPoint`, inspired by https://github.com/mdn/browser-compat-data/pull/10387#issuecomment-843982972.  Since the original `Point` API only supports 2D positioning (and is still the case with Safari's `WebKitPoint` implementation), this also adds a note to the entries for `Point` and marks them as partially implemented.

### CODE REVIEW

File: api/DOMPoint.json [hunk 1]
```diff
@@ -5,15 +5,42 @@
...
-          "edge": {
-            "version_added": "79"
-          },
+          "chrome": [
+            {
+              "version_added": "61"
+            },
+            {
+              "version_added": "2",
+              "version_removed": "39",
```
Comment (foolip) [inline api/DOMPoint.json:14]: It seems implausible that `WebKitPoint` was removed long before the replacement `DOMPoint` was available. Can you check if the data for both are really correct?

Comment (vinyldarkscratch) [reply api/DOMPoint.json:14]: As odd as this may seem, the data here is accurate...

Comment (foolip) [reply api/DOMPoint.json:14]: Oh, wait, I shouldn't be surprised by this, I'm the one who removed `WebKitPoint` in https://bugs.chromium.org/p/chromium/issues/detail?id=409956 together with `webkitConvertPointFromPageToNode` and `webkitConvertPointFromNodeToPage`.

File: api/DOMPoint.json [hunk 2]
```diff
@@ -5,15 +5,42 @@
...
+              "alternative_name": "WebKitPoint",
+              "notes": "<code>WebKitPoint</code> only supports 2D positioning using <code>x</code> and <code>y</code>."
+            }
+          ],
+          "edge": [
+            {
+              "version_added": "79"
+            },
+            {
+              "version_added": "12",
```
Comment (foolip) [inline api/DOMPoint.json:37]: Yep http://mdn-bcd-collector.appspot.com/tests/api/WebKitPoint is supported in Edge 18.

**Comment by foolip** [comment]
I've previewed this locally using https://github.com/mdn/browser-compat-data/issues/11343 and this is what it'll be:
image

It's not wrong, but I wonder if it's really the best thing to treat `WebKitPoint` as an alias of `DOMPoint` when it's a subset and was used for something quite different.

`WebKitPoint` goes together with `webkitConvertPointFromNodeToPage` and `webkitConvertPointFromPageToNode`, and those entries will eventually be removed as irrelevant if Safari removes support. When that happens, I'd want `WebKitPoint` to naturally come along for the ride, but it won't when we've baked it into the `DOMPoint` data.

So my thinking is now:
- Rename `Point`&rarr;`WebKitPoint`, `convertPointFromNodeToPage`&rarr;`webkitConvertPointFromNodeToPage` and `convertPointFromPageToNode`&rarr;`webkitConvertPointFromPageToNode` and make sure their data all match
- Remove them all when guidelines allow

@ddbeck @Elchi3 WDYT?

**Comment by Elchi3** [comment]
If `WebKitPoint` is quite a different thing and author shouldn't really bother with it for farther backwards compatibility when using `DOMPoint`, then this makes sense to me.

**Comment by vinyldarkscratch** [comment]
I've opened up #11361 which performs the changes described in the above comment!  Closing this PR now.

**Pull request closed without merging**